docs/cip/CIP-8_Matrix_Packages.md

```diff
@@ -0,0 +1,209 @@
+
+The main impact is on maintenance burden and (possible) compatibility guarantees. By creating what can be thought of as a portable serialization format for most of what is ordinarily stored in Chroma, it is likely that some properties which previously could have been considered implementation details, now will be assumed by users to be part of Chroma's public API surface.
+
+TODO examples
+
+## Proposed Change
+
+This is a proposal in two parts. First, we describe a package format. Then, we outline new chroma client capabilities and corresponding CLI subcommands that can round trip to and from this package format.
+
+Note that this CIP explicitly omits any actual dependency management behavior. While certain design choices here are motivated by preparing for that problem, directly addressing it is a problem for a future CIP.
+### Matrix Package Format
+A matrix package is a filesystem format consisting of two files, co-occurring within the same directory. Each file serves a particular, well-defined role:
+
+- `matrix.package.yaml`: defines the package name, version, and other common metadata, and certain metadata about the dataset in the package. Future iterations will add dependency declarations - the same kind of information that appears in `package.json` or `requirements.txt`. **Required**.
+- `matrix.data.parquet`: contains the package's dataset. Each row closely mirrors Chroma db content, containing original document source, computed embeddings, and metadata. **Required**, but will likely become optional in the future (more below)
+
+TODO - how do we serialize a computed index? it doesn't have the same cardinality as the data, does it? so that'd mean a separate file
+
+Most critical to note is that a matrix package contains *exactly one* dataset (see further discussion under [alternatives](#Multi-dataset%20packages.md)). For now, we can assume a 1:1 relationship between a matrix package and a Chroma collection. This simple correspondence may change once we allow dependencies between matrix packages.
+
+Some of the design choices here are oriented towards the expected eventual case of files in a matrix package smoothly coordinating to form a sparse, distributed cache. More detail on the each file follows.
+#### `matrix.package.yaml`
+
+The following is an example `matrix.package.yaml` file, which we'll use to explain each property in the file:
+
+```yaml
+name: "matrixhub.com/somegroup/examplepackage"
+version: "2023.09.08-1"
+license: "Apache-2.0"
+
+data:
+  embeddings:
+  - model: "hf://sentence-transformers/all-MiniLM-L6-v2"
+    types:
+      dimensions: 384
+      quantize:
+        bits: 8
+        method: "k-means"
+  digests:
+    data: "sha256:decafbad"
+    # index: "sha256:deadbeef" TODO do we need this?
+```
+
+##### `name`
+```yaml
+name: "matrixhub.com/somegroup/examplepackage"
+```
+The unique identifier of the package. With exactly one dataset per package, this can also be considered the name of the dataset.
```

Comment:
if a dataset equates with a single collection then perhaps the name or id of the collection?